### DISCUSSION TITLE
Upgrade for QT dependancies vulnerabilities

### DISCUSSION THREAD

**Discussion by claudiudc** [horsicq/DIE-engine#51] [Q&A] [answered]
Good day, 
First of all I want you to congratulate for this project, thank you, it is really useful for us.
Please allow me to raise a question, do you believe it would be possible to upgrade the project to target new QT > 6.1.2?
Qt5Script.dll and Qt5Core.dll have this vulnerabilities: CWE-787 CWE-787 Out-of-bounds Write: CVE-2021-38593
http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2021-38593

Thank you.

**Comment by horsicq** [comment]
Hello! Thanks a lot for the bugreport! 
Sure, I will upgrade to 6.xx

**Reply by claudiudc** [reply]
Many thanks! Really appreciate :)

**Reply by claudiudc** [reply]
Hi @horsicq 
Is it possible by any chance to provide a beta build using the new qt dll's > 6.1.2?
I appreciate your work thank you.

**Reply by horsicq** [reply]
Qt6 does not have qtscript module. I have to rewrite this part of the code.

**Reply by horsicq** [reply]
Do you need Windows or Linux build?

**Reply by claudiudc** [reply]
My apologies, I forgot to mention, we use Windows and 32 bits version of your program.
Thank you.

**Reply by horsicq** [reply]
Do you use console version or GUI?

**Reply by claudiudc** [reply]
console version

**Comment by horsicq** [comment]
Ok. I will upgrade to 6.XX anyway. But you do not really need the upgrade if you are using console version. **CVE-2021-38593** is about GUI painting(QRasterPaintEngine::fill and QPaintEngineEx::stroke).
BTW Did you try my another project NFD: https://github.com/horsicq/Nauz-File-Detector ?

**Reply by claudiudc** [reply]
Indeed, the console version does not need it, however we have a scanner at work that is giving this security warning, I will try to see if we can bypass it since we are using console version.
Also thank you for providing info for NFD, I will take a look at it.

**Reply by claudiudc** [reply]
HI @horsicq 
Please forgive me to come with an update to this topic, do you know if there Is any plan to update to new qt by any chance?
On the company where I work on the security policy require to remove qt5 dependencies from the product since our and customer scanning tools are finding qt5 vulnerabilities. 
Thank you.

**Reply by horsicq** [reply]
Hello! The next version of DiE will be with the lastest Qt6 version. I am working on it. it will be in 2-3 monthes.

**Reply by claudiudc** [reply]
Thank you for updates, really appreciate.

**Comment by horsicq** [comment, marked as answer]
Qt6 https://github.com/horsicq/DIE-engine/releases/tag/3.04

**Reply by claudiudc** [reply]
Many thanks, by any chance is possible to release 32 bit version for the one with QT 6 for windows? We are using both 32 and 64 bit windows operating systems.
Thank you.

**Comment by horsicq** [comment]
There are only 64 versions of Qt6

![image](https://user-images.githubusercontent.com/7762949/154254206-bc6b4bea-fff1-48a9-b1dc-e32c3b172e07.png)

So I keep using Qt5 for 32 projects.

![image](https://user-images.githubusercontent.com/7762949/154254521-ac33dce2-b429-4a9e-84b9-360452751e97.png)

**Reply by claudiudc** [reply]
Thank you for info.
